src/core/Table/filters/NumberRangeFilter/NumberRangeFilter.tsx

```diff
@@ -0,0 +1,86 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import { useTheme } from '../../../utils/hooks/useTheme';
+import {
+  FilterButtonBar,
+  FilterButtonBarTranslation,
+} from '../FilterButtonBar';
+import { BaseFilter } from '../BaseFilter';
+import { TableFilterProps } from '../types';
+import '@itwin/itwinui-css/css/table.css';
+import { LabeledInput } from '../../../LabeledInput';
+
+export type NumberRangeTranslation = {
+  from: string;
+  to: string;
+};
+
+const defaultStrings: NumberRangeTranslation = {
+  from: 'From',
+  to: 'To',
+};
+
+export type NumberRangeFilterProps<
+  T extends Record<string, unknown>
+> = TableFilterProps<T> & {
+  translatedLabels?: NumberRangeTranslation & FilterButtonBarTranslation;
+};
+
+export const NumberRangeFilter = <T extends Record<string, unknown>>(
+  props: NumberRangeFilterProps<T>,
+) => {
+  const { column, translatedLabels, setFilter, clearFilter } = props;
+
+  useTheme();
+
+  const translatedStrings = { ...defaultStrings, ...translatedLabels };
+
+  const [from, setFrom] = React.useState<string | undefined>(
+    column.filterValue?.[0] ?? '',
+  );
+  const [to, setTo] = React.useState<string | undefined>(
+    column.filterValue?.[1] ?? '',
+  );
+
+  const parseInputValue = (value: string | undefined) => {
+    return !value || isNaN(Number(value)) ? undefined : Number(value);
+  };
+
+  const onKeyDown = (event: React.KeyboardEvent<HTMLDivElement>) => {
+    if (event.key === 'Enter') {
+      setFilter([parseInputValue(from), parseInputValue(to)]);
+    }
+  };
+
+  return (
+    <BaseFilter>
+      <LabeledInput
+        label={translatedStrings.from}
+        value={from}
+        onChange={(e) => setFrom(e.target.value)}
+        onKeyDown={onKeyDown}
+        type='number'
+        displayStyle='inline'
+        setFocus
+      />
+      <LabeledInput
+        label={translatedStrings.to}
+        value={to}
+        onChange={(e) => setTo(e.target.value)}
+        type='number'
```

Comment:
This looks extremely ugly in dark theme 
![image](https://user-images.githubusercontent.com/9084735/119197555-f2988380-ba55-11eb-8d65-a1dfc0f06752.png)

Wonder if we should use `inputmode` instead.
https://css-tricks.com/what-to-use-instead-of-number-inputs/

Also maybe we should set `min` and `max` so that someone doesn't try to do this:
![image](https://user-images.githubusercontent.com/9084735/119197704-2e334d80-ba56-11eb-8c20-7636b4f6ec82.png)

Reply:
Don't use dark theme then 😀 I don't know from this article and the one it referenced it seems that it is a problem with number input fields that isn't just a regular numbers like credit card number.
By `min`, `max` you mean to use it inside with current input values or/and have it as props?

Reply:
min max with current input values, although that could get complicated 🤔